main.py

```diff
@@ -79,14 +81,59 @@ def get_metadata(
 
     metadata = filter_repository_by_owner(metadata, backup_only_owner_repos, token)
 
-    write_json(metadata, path_dir)
+    metadata_json = json.dumps(metadata)
+    write_metadata_in_s3_bucket(metadata_json, bucket_prefix, bucket_name)
     return metadata
 
 
-def write_json(data: Dict, path_dir: pathlib.Path):
-    file_path = path_dir / "metadata.json"
-    with open(file_path, "w") as outfile:
-        json.dump(data, outfile)
+def write_metadata_in_s3_bucket(
```

Comment:
Just a Heads Up, `write_metadata_in_s3_bucket` and `write_metadata_in_s3_bucket` functions does essentially the same thing, they write a file to s3 bucket by receiving the file buffer and a prefix name.
You could make things more reusable condensing them in a single function and then just passing in the prefix you want to use for the object.

For now let's keep it that way to ship this project, this is not a major concern anyway.